old-tutorials-in-development/tutorials-in-development/R/dc-spatio-temporal-intro/07-spatial-resolution-extent.Rmd

```diff
@@ -65,95 +95,144 @@ directory to the location of the downloaded and unzipped data subsets.
+
+``` {r vector-extent}
+#load libraries
+library(rgdal)
+library(raster)
+
+# set working directory to ensure R can find the file we wish to import and where 
+# we want to save our files. Be sure to move the unzipped download into your working directory! 
+wd <- "~/Documents/data/" #This will depend on your local environment 
+setwd(wd)
+
+# import a line shapefile
+lines_HARV <- readOGR(paste0(wd,"NEON-DS-Site-Layout-Files/HARV"), "HARV_roads")
+
+# view crs
+crs(lines_HARV)
+
+# view extent
+extent(lines_HARV)
+
+```
+
+The CRS is critical to interpreting the object extent values as it specifies 
+units. In this case the units are in meters.
+
+Learn more about vector data extent and other attributes in the NEON Data Skills Lesson<a href="https://www.neonscience.org/dc-shapefile-attributes-r" target="_blank"> *Explore Shapefile Attributes & Plot Shapefile Objects by Attribute Value in R*</a>.
 
 ### Extent in Raster Data
-The spatial extent of a raster, represents the x,y coordinates of the corners
+The spatial extent of a raster represents the "X,Y" coordinates of the corners
 of the raster in geographic space. This information, in addition to the cell
 size or spatial resolution, tells the program how to place or render each pixel
 in 2 dimensional space.  Tools like `R`, using supporting packages such as
 `rgdal`, and associated raster tools have functions that allow you to view and
 define the extent of a new raster.
 
-``` {r view-extent}
-# View the extent of the raster
-DEM@extent
+``` {r raster-extent}
+# load raster
+DSM_HARV <- raster(paste0(wd,"NEON-DS-Airborne-Remote-Sensing/HARV/DSM/HARV_dsmCrop.tif"))
 
-```
-
-<figure>
-    <a href="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/pixelDetail.png">
-    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/pixelDetail.png"></a>
-    <figcaption>The spatial resolution of a raster refers the size of each cell
-    in meters. This size in turn relates to the area on the ground that the pixel
-    represents.</figcaption>
-</figure>
+# view crs to determine units
+crs(DSM_HARV)
 
+# view the extent of the raster
+extent(DSM_HARV)
+```
 
 <figure>
-    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png">
+    <a href="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png">
+    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png"
+    alt="The image on the left has an extent of (8,8) and is less blurry than the image on the right which is stretched to an extent of (16,16)."></a>
```

Comment:
This part is a little bit confusing to me, because an `extent` object in R should have 4 cooridnates (xmin, xmax,, ymin, ymax) rather than just 2 as shown here. I think it would be more precise to either 1) list the extent at (0,8,0,8) and (0,16,0,16) or 2) just say that the images are 8 'units' wide and tall, vs 16 'units' wide and tall. (#1 would be my preference)

Reply:
Thanks for helping clarify this. I actually wasn't sure how we should write out extent.